apps/passport-server/src/services/generic-issuance/pipelines/CSVPipeline/makeGenericPODPCD.ts

```diff
@@ -0,0 +1,141 @@
+        const semaphoreIDColumnName = matchConfig.inputField;
+        // Check if the column exists
+        if (!(semaphoreIDColumnName in inputRow)) {
+          throw new Error(
+            `Could not find column "${semaphoreIDColumnName}" in CSV input data, could not match against user Semaphore ID`
+          );
+        }
+        // If the value does not match, no POD is issued
+        if (inputRow[semaphoreIDColumnName] !== requesterSemaphoreId) {
+          return undefined;
+        }
+      }
+    }
+
+    const entries: PODEntries = {};
+
+    // Iterate over configured POD entries
+    for (const [name, entryConfig] of Object.entries(output)) {
+      switch (entryConfig.source.type) {
+        // This entry is populated from configuration
+        case "configured":
+          /* @todo non-string values */
+          entries[name] = { type: "string", value: entryConfig.source.value };
+          break;
+        // This entry is populated by the email from the user's credential
+        case "credentialEmail":
+          entries[name] = { type: "string", value: requesterEmail };
+          break;
+        // This entry is populated by the Semaphore ID from the user's
+        // credential
+        case "credentialSemaphoreID":
+          entries[name] = {
+            type: "cryptographic",
+            value: BigInt(requesterSemaphoreId)
+          };
+          break;
+        // This entry is populated by some value from the CSV input.
+        case "input":
+          const columnName = entryConfig.source.name;
+          // Check if the configured column exists.
+          if (!(columnName in inputRow)) {
+            throw new Error(
+              `Could not find column "${entryConfig.source.name}" in CSV data, required by output field ${name}`
+            );
+          }
+          const value = inputRow[columnName];
+          // Create a POD entry with the appropriate type and value.
+          // This could perhaps use some more validation, possibly based on the
+          // work done in PODTicketPCD on Zod validation for POD entries.
+          if (entryConfig.type === "cryptographic") {
+            entries[name] = {
+              type: "cryptographic",
+              value: BigInt(value)
+            };
+          } else if (entryConfig.type === "int") {
+            entries[name] = {
+              type: "int",
+              value: BigInt(value)
+            };
+          } else {
+            entries[name] = {
+              type: "string",
+              value
+            };
+          }
+          break;
+      }
+    }
+
+    const pcd = await PODPCDPackage.prove({
+      entries: {
+        value: entries,
+        argumentType: ArgumentTypeName.Object
+      },
+      privateKey: {
+        value: eddsaPrivateKey,
+        argumentType: ArgumentTypeName.String
+      },
+      id: {
+        value: uuidv5(Object.values(inputRow).join(","), pipelineId),
```

Comment:
This is interesting.  Based on my Googling, uuidv5 is unique based on a hash of the input, which in this case is the entry values (but not their names, which seems an interesting choice).

This means if the issuer changes a field in the CSV, it gets issued as a new POD?  I.e. there's no support for reissuing the "same" POD to replace it in Zupass with updated data, as we do with tickets.

That seems like a feature people will want, which further suggests there needs to be some way to attach stable IDs to rows, either automatically or via a direct user choice.

Reply:
Ideally the Prove function would provide a way to auto-generate a PCD ID from the POD's contentID.  I'll make a task for that.

Reply:
https://linear.app/0xparc-pcd/issue/0XP-1010/stable-id-for-podpcd